File: csharp/ToolGood.Algorithm/Internals/Functions/MathTrigonometric/Function_ACOT.cs
using System;
using System.Collections.Generic;
using ToolGood.Algorithm.Enums;

namespace ToolGood.Algorithm.Internals.Functions.MathTrigonometric
{
	internal sealed class Function_ACOT : Function_1
	{
		public Function_ACOT(FunctionBase[] funcs) : base(funcs)
		{
			if (funcs.Length != 1) {
				throw new ArgumentException($"Function '{Name}' requires exactly 1 parameter.");
			}
		}

		public override string Name => "Acot";

		public override Operand Evaluate(AlgorithmEngine engine, Func<AlgorithmEngine, string, Operand> tempParameter)
		{
			var args1 = GetNumber_1(engine, tempParameter);
			if (args1.IsErrorOrNone) { return args1; }
			return Operand.Create(MathEx.PI / 2 - MathEx.Atan(args1.NumberValue));
		}
		public override OperandType GetResultType()
		{
			return OperandType.NUMBER;
		}

		internal override void GetParameterTypes(NoneEngine noneEngine, List<ParameterType> result, OperandType operandType, string op = null, string val = null)
		{
			func1.GetParameterTypes(noneEngine, result, OperandType.NUMBER);
		}
	}
}

File: JavaScript/src/Internals/Functions/MathTrigonometric/Function_ACOT.js
import { Function_1 } from '../Function_1.js';
import { Operand } from '../../../Operand.js';

class Function_ACOT extends Function_1 {
    get Name() {
        return "Acot";
    }

    constructor(func1) {
        super(func1);
    }

    evaluate(engine, tempParameter) {
        let args1 = this.getNumber_1(engine, tempParameter);
        if (args1.IsError) { return args1; }
        return Operand.Create(Math.PI / 2 - Math.atan(args1.NumberValue));
    }
}

export { Function_ACOT };